api/versions.go

```diff
@@ -427,15 +444,18 @@ func (api *DatasetAPI) updateVersion(ctx context.Context, body io.ReadCloser, ve
 			log.Error(ctx, "putVersion endpoint: failed to update version document", err, data)
 			return err
 		}
+		log.Info(ctx, "DEBUG UpdateVersion to MongoDB", log.Data{"time": time.Since(t0), "etag": eTag})
 
 		return nil
 	}
 
 	// acquire instance lock to prevent race conditions on instance collection
+	t0 = time.Now()
 	lockID, err := api.dataStore.Backend.AcquireInstanceLock(ctx, currentVersion.ID)
 	if err != nil {
 		return nil, nil, nil, err
 	}
+	log.Info(ctx, "DEBUG AcquireInstanceLock", log.Data{"time": time.Since(t0)})
 	defer api.dataStore.Backend.UnlockInstance(ctx, lockID)
```

Comment:
Please can you add code to show how long it takes to do: UnlockInstance() (at ~ line 459)
As thats the one that took 40ms+ elsewhere